src/main/kotlin/org/jetbrains/bio/experiments/fit/Span2PeakCallingExperiment.kt

```diff
@@ -0,0 +1,169 @@
+        paths: SpanDataPaths,
+        mapabilityPath: Path?,
+        fragment: Fragment,
+        binSize: Int,
+        modelFitter: Fitter<Model>,
+        modelClass: Class<Model>,
+        states: Array<State>,
+        nullHypothesis: NullHypothesis<State>,
+        unique: Boolean = true,
+        fixedModelPath: Path? = null
+) : SpanModelFitExperiment<Model, State>(
+    genomeQuery to createDataQuery(genomeQuery, paths, mapabilityPath, fragment, binSize, unique),
+    listOf(paths), mapabilityPath,
+    listOfNotNull("y", paths.control?.let { "input" }),
+    fragment, binSize,
+    modelFitter, modelClass,
+    states, nullHypothesis,
+    unique,
+    fixedModelPath
+) {
+
+    override val modelPath: Path get() = fixedModelPath ?: experimentPath / "$id.span2"
+    override val id = reduceIds(
+        listOfNotNull(paths.treatment, paths.control, mapabilityPath).map { it.stemGz } +
+                listOfNotNull(fragment.nullableInt, binSize).map { it.toString() }
+    )
+
+    companion object {
+
+        fun binnedCoverageAsInt(chr: Chromosome, coverage: Coverage, binSize: Int): IntArray {
+            val len = (chr.length - 1) / binSize + 1
+            val res = IntArray(len)
+            for (i in 0 until len - 1) {
+                res[i] = coverage.getBothStrandsCoverage(
+                    ChromosomeRange(i * binSize, (i + 1) * binSize, chr)
+                )
+            }
+            res[len - 1] = coverage.getBothStrandsCoverage(
+                ChromosomeRange((len - 1) * binSize, chr.length, chr)
+            )
+            return res
+        }
+
+        fun binnedCoverageAsDouble(chr: Chromosome, coverage: Coverage, binSize: Int): DoubleArray {
+            val len = (chr.length - 1) / binSize + 1
+            val res = DoubleArray(len)
+            for (i in 0 until len - 1) {
+                res[i] = coverage.getBothStrandsCoverage(
+                    ChromosomeRange(i * binSize, (i + 1) * binSize, chr)
+                ).toDouble()
+            }
+            res[len - 1] = coverage.getBothStrandsCoverage(
+                ChromosomeRange((len - 1) * binSize, chr.length, chr)
+            ).toDouble()
+            return res
+        }
+
+        fun meanGC(chr: Chromosome, binSize: Int): DoubleArray {
+            val len = (chr.length - 1) / binSize + 1
+            val seq = chr.sequence
+            val gcContent = DoubleArray(len)
+            for (i in 0 until len - 1) {
+                gcContent[i] = seq.substring(i * binSize, (i + 1) * binSize)
+                        .count { it == 'c' || it == 'g' }.toDouble() / binSize
+            }
+            gcContent[len - 1] = seq.substring((len - 1) * binSize, seq.length)
+                    .count { it == 'c' || it == 'g' }
+                    .toDouble() / (seq.length - (len - 1) * binSize)
+            return gcContent
+        }
+
+        fun binnedMapability(chr: Chromosome, mapabilityPath: Path, binSize: Int): DoubleArray {
```

Comment:
Probably we should move this to `Span2Covariates` class, because it has nothing to do with model fitting itself.

Reply:
And what about simple test case?

Reply:
It has to do with data preparation for the fitting (`createDataQuery`). For example, the code that fills a dataframe with binned coverage scores resides in `SpanModelFitExperiment`.
On the other hand, it would probably be nice to move this logic to a dedicated class.

Reply:
Moved the dataframe generation logic to `SpanFitInformation`'s `dataQuery`.

Reply:
To provide a test case, we would need to add a bigWig file for `to1` genome to the repository. I will ask @iromeo for the best way to generate it.

Reply:
Added mapability to test organism data generation, and a test on `binnedMapability` method.